firehose.c

```diff
@@ -390,6 +395,8 @@ static int firehose_try_configure(struct qdl_device *qdl, bool skip_storage_init
 				break;
 			}
 		}
+
+		free(buf);
```

Comment:
I think replacing the malloc() with alloca() would be cleaner, then we don't have to explicitly free the memory (and we don't need the NULL check...)